pip/pep425tags.py

```diff
@@ -30,7 +30,48 @@ def get_abbr_impl():
 
 def get_impl_ver():
     """Return implementation version."""
-    return ''.join(map(str, sys.version_info[:2]))
+    try:
+        impl_ver = sysconfig.get_config_var("py_version_nodot")
+    except IOError as e:  # Issue #1074
+        warnings.warn("{0}".format(e), RuntimeWarning)
+        impl_ver = None
+    if not impl_ver or get_abbr_impl() == 'pp':
+        impl_ver = ''.join(map(str, get_impl_version_info()))
+    return impl_ver
+
+
+def get_impl_version_info():
+    """Return sys.version_info-like tuple for use in decrementing the minor
+    version."""
+    if get_abbr_impl() == 'pp':
+        # as per https://github.com/pypa/pip/issues/2882
+        return (sys.version_info[0], sys.pypy_version_info.major,
+                sys.pypy_version_info.minor)
+    else:
+        return sys.version_info[0], sys.version_info[1]
+
+
+def get_abi_tag():
+    """Return the ABI tag based on SOABI (if available) or emulate SOABI
+    (CPython 2, PyPy)."""
+    try:
+        soabi = sysconfig.get_config_var('SOABI')
+    except IOError as e:  # Issue #1074
+        warnings.warn("{0}".format(e), RuntimeWarning)
+        soabi = None
+    impl = get_abbr_impl()
+    if not soabi and impl in ('cp', 'pp'):
+        d = 'd' if hasattr(sys, 'pydebug') and sys.pydebug else ''
+        m = 'm' if impl == 'cp' else ''
```

Comment:
Sorry, that bit came from Nick's comments [here](https://bitbucket.org/pypa/wheel/issues/63/backport-soabi-to-python-2). `sysconfig.get_config_var('WITH_PYMALLOC')` does indeed seem to work, updating shortly.